hera_cal/redcal.py

```diff
@@ -789,3 +837,335 @@ def is_redundantly_calibratable(antpos, bl_error_tol=1.0):
+    solar_alts = utils.get_sun_alt(hd.times, latitude=lat, longitude=lon)
+    solar_flagged = solar_alts > solar_horizon
+    if verbose and np.any(solar_flagged):
+        print(len(hd.times[solar_flagged]), 'integrations flagged due to sun above', solar_horizon, 'degrees.')
+    
+    # loop over data, performing partial loading 
+    for pols in pol_load_list:
+        if verbose:
+            print('Now calibrating', pols, 'polarization(s)...')
+        reds = filter_reds(all_reds, ex_ants=ex_ants, pols=pols)
+        for tinds in np.split(np.arange(nTimes)[~solar_flagged], 
+                              np.arange(nInt_to_load, len(hd.times[~solar_flagged]), nInt_to_load)):
+            if len(tinds) > 0:
+                if verbose:
+                    print('    Now calibrating times', hd.times[tinds[0]], 'through', hd.times[tinds[-1]], '...')
+                data, _, _ = hd.read(times=hd.times[tinds], polarizations=pols)
+                cal = redundantly_calibrate(data, reds, freqs=hd.freqs, times_by_bl=hd.times_by_bl, 
+                                            conv_crit=conv_crit, maxiter=maxiter, 
+                                            check_every=check_every, check_after=check_after, gain=gain)
+                # gather results 
+                for ant in cal['g_omnical'].keys():
+                    rv['g_firstcal'][ant][tinds, :] = cal['g_firstcal'][ant]
+                    rv['gf_firstcal'][ant][tinds, :] = cal['g_firstcal'][ant]
+                    rv['g_omnical'][ant][tinds, :] = cal['g_omnical'][ant]
+                    rv['gf_omnical'][ant][tinds, :] = cal['gf_omnical'][ant]
+                    rv['chisq_per_ant'][ant][tinds, :] = cal['chisq_per_ant'][ant]
+                for bl in cal['v_omnical'].keys():
+                    rv['v_omnical'][bl][tinds, :] = cal['v_omnical'][bl]
+                    rv['vf_omnical'][bl][tinds, :] = cal['vf_omnical'][bl]
+                if pol_mode in ['1pol', '2pol']:
+                    for antpol in cal['chisq'].keys():
+                        rv['chisq'][antpol][tinds, :] = cal['chisq'][antpol]
+                else:  # duplicate chi^2 into both antenna polarizations
+                    for antpol in rv['chisq'].keys():
+                        rv['chisq'][antpol][tinds, :] = cal['chisq']
+
+    return rv
+
+
+def redcal_run(input_data, firstcal_suffix='.first.calfits', omnical_suffix='.omni.calfits', omnivis_suffix='.omni.vis', 
+               outdir=None, ant_metrics_file=None, clobber=False, nInt_to_load=8, pol_mode='2pol', ex_ants=[], ant_z_thresh=4.0, 
+               solar_horizon=0.0, conv_crit=1e-10, maxiter=500, check_every=10, check_after=50, gain=.4,
+               append_to_history='', verbose=False):
+    '''Perform redundant calibration (firstcal, logcal, and omnical) an entire HERAData object, loading only 
+    nInt_to_load integrations at a time and skipping and flagging times when the sun is above solar_horizon.
```

Comment:
Emphasize in comments that this does iterative antenna flagging.  We probably want to be able to run this function without partial data loading on existing data as well.  Can we make nInt_to_load default to -1 (all) and if so, uses the redundantly_calibrate instead of redcal_partial_io_iteration?

Reply:
We made a decision a while ago that we want to break backwards compatibility for non-uvh5 data. This is all based on new code in hera_cal.io that operates differently (and better) on uvh5 files. Paul already converted all of IDR2 data to uvh5 and we can do it with the rest of IDR3 when we're ready for the big run.

Letting redcal_run do an end-run around redcal_partial_io_iteration and just use redundantly_calibrate is not easy at all and I'm not sure what the use case is.

I'm happy to change the default nint_to_load, but I figured it would be more helpful to have it be a reasonable value.

Reply:
Also, I forgot to update that docstring. I've fixed that.